src/api/composition/CompositionProvider.js

```diff
@@ -96,8 +96,9 @@ export default class CompositionProvider {
    * object.
    * @param {DomainObject} domainObject the domain object
    *        for which to load composition
-   * @returns {Promise<Identifier[]>} a promise for
-   *          the Identifiers in this composition
+   * @returns {Promise<Identifier[] | DomainObject[]>} a promise for
```

Comment:
Should include identifiers as keyStrings as well

Reply:
I took a slightly different approach here. Our API previously said that Composition Providers return Identifiers. Indeed the DefaultCompositionProvider also claimed to only return identifiers. So, supporting keystrings would technically require a change to our API interface, and keystrings are deprecated. 

So, instead I've modified the DefaultCompositionProvider to [guarantee that only identifiers are returned](https://github.com/nasa/openmct/pull/8037/files#diff-bb4c368d5df65905cfd33d9f8197ab096456a8c8fcd2b4ff043c7dd2b3d4fa95R78), and removed the check for keystrings that I had added to the CompositionCollection.